usr.bin/mdo/mdo.c

```diff
@@ -22,10 +23,27 @@ usage(void)
 	exit(EXIT_FAILURE);
 }
 
+static char*
+join_strings(char **arr, int size, const char *delimiter)
```

Comment:
Since this function is only used once, it can be folded into `main()`

Reply:
I think the separation would make the code easier to read and without loss the original code logic?

Reply:
I don't really have strong opinion here, it's more of a personal preference.

However, this approach also matches existing patterns of other similar situations: from readability point of view, while separating a more complex function is generally helpful (e.g. this program would do some complicated operations A, B, and C in order, and these operations are implemented as separate functions), it can actually hurt readability to overly wrap simple operations.  My personal rule of thumb is to not extract an operation into a function if a) it's only used once, and b) the function itself plus the caller's code block would fit one page or screen (about 20-25 lines) worth of code block.

Note that there are other similar situations where it's done in place instead of in a separate function, for example in `apply(1)` and `newsyslog(8)`.